addons/optics/script_component.hpp

```diff
@@ -19,6 +19,32 @@
 
 #define PARSE(value) (call compile format ["%1", value])
 #define AMBIENT_BRIGHTNESS (sunOrMoon * sunOrMoon * (1 - overcast * 0.25) + moonIntensity / 5 * (1 - overcast) min 1) // idea by Falke
+#define WEAPON_MAGAZINES(unit,weapon) (weaponsItems (unit) select {_x select 0 == (weapon)} param [0, []] select {_x isEqualType []})
+
+// addWeapon, but without leeching magazines
+#define ADD_GUN(unit,gun) (unit) setUnitLoadout [\
```

Comment:
`setUnitLoadout` breaks anything that uses unique mag IDs
ref: https://github.com/CBATeam/CBA_A3/blob/master/addons/common/fnc_getMagazineIndex.sqf

Reply:
What alternative do we have? These macros are currently only used by the G36A1 and G36A2 and variants.